### PR TITLE
Onyx One-piece

### PR DESCRIPTION
**Pull request by Thrunic**
Onyx is ready for review. It's feature complete, I've written the documentation, and I think it's ready to be tested.

### CODE REVIEW

**Comment by joostdecock** [comment]
Can you address the eslint errors @Thrunic ?

**Comment by Thrunic** [comment]
> Can you address the eslint errors @Thrunic ?

This one works now: ESLint Changed Files / lint (18.x) (pull_request) Successful in 1m

ESLint All Files / lint (18.x) still fails on files that are not part of Onyx.

**Review by joostdecock — CHANGES_REQUESTED** [review]
This is great stuff @Thrunic 
There's a lot of repeating things that I pointed out, so all in all it's just a couple of issues.
However, people tend to copy/paste from other designs, so it's important to do things the best way so that others don't copy bad habits.

File: designs/onyx/src/back.mjs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        .join(paths.hemBase.offset(options.legRibbing ? sa : absoluteOptions.legHem))
+        .join(paths.saBase.offset(sa))
+        .attr('class', 'fabric sa')
+      if (options.backOnFold) {
+        paths.sa.line(points.cfNeck)
+        paths.sa2 = new Path()
+          .move(points.cfCrotch)
+          .join(paths.inseamBase.offset(sa))
+          .line(points.inseamHem.shift(270, options.legRibbing ? sa : absoluteOptions.legHem))
+          .attr('class', 'fabric sa')
```
[the remark this answers is not in the document]
Comment (Thrunic) [reply designs/onyx/src/back.mjs:181]: Is there a difference between 'sa fabric' and 'fabric sa'?

File: designs/onyx/src/raglansleeve.mjs [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+  return part
+}
+
+export const raglanSleeve = {
+  name: 'onyx.raglanSleeve',
+  after: [front, back],
+  plugins: [],
+  draft: draftRaglanSleeve,
+  measurements: ['neck', 'chest', 'biceps', 'wrist', 'shoulderToWrist'],
```
[the remark this answers is not in the document]
Comment (Thrunic) [reply designs/onyx/src/raglansleeve.mjs:317]: 'shoulderToWrist', 'biceps', and 'wrist' aren't declared elsewhere.

File: designs/onyx/src/hood.mjs [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+
+  return part
+}
+
+export const hood = {
+  name: 'onyx.hood',
+  plugins: [],
+  draft: draftHood,
+  after: [front, back, raglanSleeve],
+  measurements: ['neck', 'chest', 'biceps', 'wrist', 'head'],
```
[the remark this answers is not in the document]
Comment (Thrunic) [reply designs/onyx/src/hood.mjs:246]: 'head' isn't declared elsewhere.

File: designs/onyx/src/zipperguard.mjs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  }
+
+  return part
+}
+
+export const zipperGuard = {
+  name: 'onyx.zipperGuard',
+  plugins: [],
+  draft: draftZipperGuard,
+  after: [front, neckband],
```
[the remark this answers is not in the document]
Comment (Thrunic) [reply designs/onyx/src/zipperguard.mjs:101]: Everything it needs is actually just in base, so I changed it to after: [base], and changed the imports on top.

[the remark this answers is not in the document]
Comment (Thrunic) [reply designs/onyx/src/zipperguard.mjs:101]: Everything it needs is actually just in base, so I changed it to after: [base], and changed the imports on top.

File: designs/onyx/src/zipperguard.mjs [hunk 5]
```diff
@@ -0,0 +1,125 @@
+import { neckband } from './neckband.mjs'
+import { front } from './front.mjs'
```
[the remark this answers is not in the document]
Comment (Thrunic) [reply designs/onyx/src/zipperguard.mjs:2]: Both imports replaced by base from ./base.mjs

**Comment by Thrunic** [comment]
@joostdecock I've fixed the issues noted in your comments. There's a few that I'm not sure about, so I put replies in those messages and left them unresolved.

I'm also pretty sure there's going to be a bunch unneeded lines of code, mostly in the function draftPartName declarations. I'm going to run tests and get rid of them.

Lastly, I noticed that I neglected to make a zipperPosition option (list option, possible values of front (default), back, none). I'm going to add that in, and make the needed changes of the code to draw the zipper on the correct piece.

I'll post another message when the code is ready for further review.

**Comment by Thrunic** [comment]
@joostdecock Onyx should be ready for further review. I've made all requested changes, along a few other small fixes/additions.

There were several changes that I felt were missing something. I've left a comment on those conversations and fixed them with what I believe is correct. If there are still errors, please let me know.

Thank you again for all your hard work.

**Review by joostdecock — APPROVED** [review]

**Pull request merged**